### PR TITLE
Enable remote tool connections for the DVM. 

### PR DESCRIPTION
**Pull request by rhc54**
Fix notifications so we "de-bounce" termination calls

Signed-off-by: Ralph Castain <rhc@open-mpi.org>

### CODE REVIEW

**Comment by rhc54** [comment]
@andre-merzky

**Comment by rhc54** [comment]
@andre-merzky I decided not to pursue the automated "probe" as I just don't have the time right now. However, I did implement the prior "report-uri" method. Please let me know how it goes for you.

**Comment by ibm-ompi** [comment]
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/1d3584763c20267f4974019d7a6358d2

File: orte/tools/prun/prun.c [hunk 1]
```diff
@@ -184,7 +193,7 @@ static void evhandler(int status,
     }
     if (!fired) {
         fired = true;
-        ORTE_ACTIVATE_PROC_STATE(ORTE_PROC_MY_NAME, ORTE_PROC_STATE_TERMINATED);
+        ORTE_ACTIVATE_PROC_STATE(ORTE_PROC_MY_NAME, 123456);
```
Comment (andre-merzky) [inline orte/tools/prun/prun.c]: Is this intentional?

**Comment by andre-merzky** [comment]
Yep, great - I'll give that a try!

**Comment by ibm-ompi** [comment]
The IBM CI (XL Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/d6ba4959f60e5e167ab37e6bccc9a8bb

**Comment by ibm-ompi** [comment]
The IBM CI (PGI Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/f7ebcfd0320766192ca10964effbff66

**Comment by rhc54** [comment]
@andre-merzky thanks for flagging that debug code - I have removed it

**Comment by rhc54** [comment]
@andre-merzky I should have mentioned: the cmd line to allow remote tools to operate looks like this:
```
$ orte-dvm --report-uri foo.txt --remote-tools
```

**Comment by andre-merzky** [comment]
This looks good from my end - but passing the URI seems to only work as file:

```
thinkie  merzky  …/ompi/src/ompi.rhc54  [topic/remote] $ cat foo.txt 
3834118144.0;tcp4://10.0.0.4:52125
v2.1.0

thinkie  merzky  …/ompi/src/ompi.rhc54  [topic/remote] $ prun --hnp file:foo.txt -np 2 date 
[thinkie:21753] JOB COMPLETED WITH STATUS 0
[thinkie:21753] JOB COMPLETED WITH STATUS 0
Sat Oct  7 00:00:11 CEST 2017
Sat Oct  7 00:00:11 CEST 2017
[thinkie:21753] JOB [58504,7] EXECUTING
[thinkie:21753] JOB COMPLETED WITH STATUS 0

thinkie  merzky  …/ompi/src/ompi.rhc54  [topic/remote] $ prun --hnp '3834118144.0;tcp4://10.0.0.4:52125' -np 2 date             
[thinkie:21822] ptl_tcp_parse_uri: Could not convert 144.0;tcp4
[thinkie:21822] PMIX ERROR: BAD-PARAM in file ptl_tcp.c at line 621
[thinkie:21822] OPAL ERROR: Unreachable in file pmix2x_client.c at line 240
[thinkie:21822] [[57724,0],0] ORTE_ERROR_LOG: Unreachable in file base/ess_base_std_tool.c at line 193
...

thinkie  merzky  …/ompi/src/ompi.rhc54  244  [topic/remote] $ prun --hnp 'tcp4://10.0.0.4:52125' -np 2 date 
[thinkie:21887] OPAL ERROR: Unreachable in file pmix2x_client.c at line 240
[thinkie:21887] [[57661,0],0] ORTE_ERROR_LOG: Unreachable in file base/ess_base_std_tool.c at line 193
...
```

But the file version is good enough in our case - so thanks! :)

**Comment by rhc54** [comment]
huh - ok, i'll take a run at it

**Pull request merged**